docgen/src/guide/Server-side_rendering.md

```diff
@@ -6,8 +6,57 @@ category: guide
 navWeight: 30
 ---
 
-React InstantSearch is not yet compatible with server-side rendering. This is one of our priority topics
-right now. We will soon provide API entries to handle this usecase.
+React InstantSearch builds the Algolia query based on every widgets declared. That's why at the first rendering there is no results yet. This is inconvenient when doing server side rendering as the html generated by the server will not contain any hits. 
+
+If you want to have a server side rendering application using React InstantSearch you will need to use the special API entries we provide for this particular use-case: 
+
+* a new type of `<InstantSearch>` available under `'react-instantsearch/server'`
+* a `findResults` function available under `'react-instantsearch/server'`
+
+Let's see how it works!
+
+First you will need to get the first results using the `findResults` function. This function takes two parameters: 
+
+* Your <InstantSearch/> instance
+* A parameters object containing a `searchState`. This is optional and useful only you are passing a `searchState` to your `<InstantSearch/>` instance. 
+
+```js
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Hits, } from 'react-instantsearch/dom';
+import { InstantSearch } from 'react-instantsearch/server';
+import { findResults } from 'react-instantsearch/server';
+
+class App extends React.Component {
+  static propTypes = {
+    searchState: PropTypes.object,
+    resultsState: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
+    onSearchStateChange: PropTypes.func,
+  };
+
+  render() {
+    return (
+      <InstantSearch
+        appId="latency"
+        apiKey="6be0576ff61c053d5f9a3225e2a90f76"
+        indexName="ikea"
+        resultsState={this.props.resultsState}
+        onSearchStateChange={this.props.onSearchStateChange}
+        searchState={this.props.searchState}
+      >
+        <Hits />
+      </InstantSearch>
+    );
+  }
+}
+
+const resultsState = await findResults(App, { searchState });
+```
+
+Then, once you'll get those results, you can actually give it as a prop to your `<InstantSearch/>` instance. You'll need a mechanism to render your app after fetching those results. 
```

Comment:
I think in the end, I should remove the code sample. Right now, it's partial so not very useful, and if I go for the full one then:

- it's next.js only
- it's complex
- it will be static and as soon as the next-app demo will update, I'm sure it will be forgotten. 

Maybe that the link to the next.js example is enough. What do you think?